versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/CommitRetry.java

```diff
@@ -60,6 +61,10 @@ static <T> T commitRetry(Persist persist, CommitAttempt<T> attempt, TryLoopState
           throw new RetryTimeoutException(i, tls.currentNanos() - t0);
         }
         retryState = e.retryState();
+      } catch (UnknownOperationResultException e) {
+        if (!tls.retry(t1)) {
+          throw new RetryTimeoutException(i, tls.currentNanos() - t0);
```

Comment:
Does this retry with a different commit timestamp? (I guess so, just wanted to confirm.)

Reply:
Logically no (`CommitMetadata` is supplied externally) - but the underlying `CommitObj` is created for every attempt.

Reply:
But that's not OK, is it? If the first commit attempt actually succeeded, but `UnknownOperationResultException` was thrown, then attempting to store the same object again would necessarily result in `Persist.storeObj()` returning false.

Reply:
Could we have a test for this (I also commented on that in another thread)?

Reply:
> But that's not OK, is it? If the first commit attempt actually succeeded, but UnknownOperationResultException was thrown, then attempting to store the same object again would necessarily result in Persist.storeObj() returning false.

These cases are all handled in `CommitImpl.commit()` and tested via `TestVersionStoreImpl.commitWithDatabaseTimeout()`.

Reply:
Sorry if I sounded overly harsh - Just didn't understand that you meant something different.